src/interned.rs

```diff
@@ -213,13 +244,19 @@ where
         self.ingredient_index
     }
 
-    fn maybe_changed_after(
-        &self,
-        _db: &dyn Database,
-        _input: Option<Id>,
-        revision: Revision,
-    ) -> bool {
-        revision < self.reset_at
+    fn maybe_changed_after(&self, db: &dyn Database, input: Id, revision: Revision) -> bool {
+        let value = db.zalsa().table().get::<Value<C>>(input);
+        if value.first_interned_at > revision {
+            // The slot was reused.
+            return true;
+        }
+
+        if value.first_interned_at < revision {
+            // The slot is valid in this revision but we have to sync the value's revision.
+            value.last_interned_at.store(db.zalsa().current_revision());
+        }
```

Comment:
I already discussed this with @ibraheemdev but I document it here in case anyone else ends up reviewing this PR or tries to fix the failing test. 

The fix here is to always set `last_interned_at` even then the `first_interned_at` is equal to `revision`
```suggestion
        if value.first_interned_at > revision {
            // The slot was reused.
            true
        } else {
            // The slot is valid in this revision but we have to sync the value's revision.
            value.last_interned_at.store(db.zalsa().current_revision());
            false
        }
```